pkg/url/url.go

```diff
@@ -85,39 +86,68 @@ func Get(client *occlient.Client, localConfig *config.LocalConfigInfo, urlName s
 	return URL{}, errors.New(fmt.Sprintf("the url %v does not exist", urlName))
 }
 
-// GetIngress returns ingress spec for given URL name
-func GetIngress(kClient *kclient.Client, envSpecificInfo *envinfo.EnvSpecificInfo, urlName string, componentName string) (URL, error) {
+// GetIngressOrRoute returns ingress/route spec for given URL name
+func GetIngressOrRoute(client *occlient.Client, kClient *kclient.Client, envSpecificInfo *envinfo.EnvSpecificInfo, urlName string, componentName string, routeSupported bool) (URL, error) {
 	remoteExist := true
+	var ingress *iextensionsv1.Ingress
+	var route *routev1.Route
+	var getRouteErr error
 	// Check whether remote already created the ingress
-	ingress, err := kClient.GetIngress(urlName)
-	if kerrors.IsNotFound(err) {
+	ingress, getIngressErr := kClient.GetIngress(urlName)
+	if kerrors.IsNotFound(getIngressErr) && routeSupported {
+		// Check whether remote already created the route
+		route, getRouteErr = client.GetRoute(urlName)
+	}
+	if kerrors.IsNotFound(getIngressErr) && kerrors.IsNotFound(getRouteErr) {
```

Comment:
This will fail when routeSupported is false as `getRouteErr` is nil

Reply:
Thanks for catching this bug. 
Updated the condition, and also add unit tests for describing ingresses if route is not supported.

Reply:
Cool :+1: